templates/glossary.md

```diff
@@ -351,7 +373,20 @@ Commands such as `have`, `suffices`, and `show` can be used to write structured
 
 ### TPIL
 
-"[Theorem Proving in Lean](https://leanprover.github.io/theorem_proving_in_lean/index.html)", a free online textbook by Jeremy Avigad, Leonardo de Moura, and Soonho Kong, "designed to teach you to develop and verify proofs in Lean".  Starting with a simple introduction to the [type theory](https://leanprover.github.io/theorem_proving_in_lean/dependent_type_theory.html) used in Lean, the text proceeds to explain topics such as [tactics](https://leanprover.github.io/theorem_proving_in_lean/tactics.html), [inductive types](https://leanprover.github.io/theorem_proving_in_lean/inductive_types.html), and [type classes](https://leanprover.github.io/theorem_proving_in_lean/type_classes.html).
+"[Theorem Proving in Lean](https://leanprover.github.io/theorem_proving_in_lean/index.html)", a free online textbook by Jeremy Avigad, Leonardo de Moura, and Soonho Kong, "designed to teach you to develop and verify proofs in Lean".
+Starting with a simple introduction to the [type theory](https://leanprover.github.io/theorem_proving_in_lean/dependent_type_theory.html) used in Lean, the text proceeds to explain topics such as [tactics](https://leanprover.github.io/theorem_proving_in_lean/tactics.html), [inductive types](https://leanprover.github.io/theorem_proving_in_lean/inductive_types.html), and [type classes](https://leanprover.github.io/theorem_proving_in_lean/type_classes.html).
+
+### type theory
+
+A formal system with two fundamental kinds of objects, [terms](#term) and types.
+It also may refer to the field of study of such languages, as there are nuances between various additional properties or features a specific type theory may contain.
+
+Lean's [dependent](#dependent-type-theory) type theory underlies its foundational system of mathematics in contrast to set theory.
```

Comment:
I'm not 100% happy with getting to the definition here via having to start with set theory before getting to the type theory definition but I couldn't think of a better way to describe this understandably to someone otherwise (certainly it didn't really click for me personally until seeing something like the below example). Ideas welcome for how to reorder though if someone has one.

Reply:
I think it's a reasonable approach.